File: jar/src/main/java/jline/api/measures/Ms_intersection.java
package jline.api.measures;

import jline.util.matrix.Matrix;

public final class Ms_intersection {
    private Ms_intersection() {}

    /**
     * Intersection distance between two probability distributions.
     * Part of the intersection family.
     *
     * @param P first probability distribution
     * @param Q second probability distribution
     * @return Intersection distance
     */
    public static double ms_intersection(Matrix P, Matrix Q) {
        if (P.length() != Q.length()) {
            throw new IllegalArgumentException("Distributions must have the same size");
        }

        double sum = 0.0;
        for (int i = 0; i < P.length(); i++) {
            sum += Math.abs(P.get(i) - Q.get(i));
        }

        return 0.5 * sum;
    }
}

File: jar/src/main/kotlin/jline/api/measures/Ms_intersection.kt
package jline.api.measures

import jline.util.matrix.Matrix
import kotlin.math.abs

/**
 * Intersection distance between two probability distributions.
 * Part of the intersection family.
 * 
 * @param P first probability distribution
 * @param Q second probability distribution
 * @return Intersection distance
 */
fun ms_intersection(P: Matrix, Q: Matrix): Double {
    require(P.length() == Q.length()) { "Distributions must have the same size" }
    
    var sum = 0.0
    for (i in 0 until P.length()) {
        sum += abs(P.get(i) - Q.get(i))
    }
    
    return 0.5 * sum
}
/**
 * Intersection metric algorithms
 */
@Suppress("unused")
class MsIntersectionAlgo {
    companion object {
        // Class documentation marker for Dokka
    }
}